### DISCUSSION TITLE
$RefreshSig$ is not defined error after using tutorial

### DISCUSSION THREAD

**Discussion by tonekk** [shakacode/react_on_rails#1554] [Q&A] [answered]
Hi,

I just created a new rails app following the tutorial.
When running `foreman start -f Procfile.dev-static`, I get the following errors when going to `http://localhost:3000/hello_world`:
(Btw I had to change `Procfile.dev-static` to:
```
# You can run these commands in separate shells
web: rails s -p 3000

# Next line runs a watch process with webpack to compile the changed files.
# When making frequent changes to client side assets, you will prefer building webpack assets
# upon saving rather than when you refresh your browser page.
# Note, if using React on Rails localization you will need to run
# `bundle exec rake react_on_rails:locale` before you run bin/webpacker
webpack: sh -c 'rm -rf public/packs/* || true && bin/shakapacker -w'
```
as it used `webpacker` before.)

<img width="645" alt="Screen Shot 2023-07-09 at 22 20 59" src="https://github.com/shakacode/react_on_rails/assets/3327396/2bb4bf4d-10ab-4060-9df2-0547772084ae">

When running with `rails s` and `shakapacker-dev-server`, I get:

<img width="644" alt="Screen Shot 2023-07-09 at 22 25 58" src="https://github.com/shakacode/react_on_rails/assets/3327396/8df7366a-6a0a-46f8-912a-9622169dcb62">

How to solve this?

Thanks 🙏

**Comment by justin808** [comment]
@ahangarha is working on updates.

**Comment by Judahmeek** [comment, marked as answer]
@tonekk I think this was user error... Kinda, anyway...

The `$RefreshSig$ is not defined` error you have is caused when the `react-refresh` babel plugin is used, but a corresponding Webpack plugin is not.

The logic the ReactOnRails installation generator adds is that [the react-refresh babel plugin is used if the WEBPACK_SERVE environment variable is defined](https://github.com/shakacode/react_on_rails/blob/master/lib/generators/react_on_rails/templates/base/base/babel.config.js.tt#L19), which is usually only [if webpack-dev-server is running](https://github.com/webpack/webpack-dev-server/pull/3000/files#diff-b706bbadb3d2315d99678f05e1f4bfd78a4bb4a7a6c65f3e81f1386fee3c883aR37).

[The Webpack plugin](https://github.com/shakacode/react_on_rails/blob/master/lib/generators/react_on_rails/templates/base/base/config/webpack/development.js.tt#L9), however, is only added if [Shakapacker determines that the dev_server is running (which it checks by websocket) & that you've configured CSS to be inlined.](https://github.com/shakacode/shakapacker/blob/94f47bc8bed39db1e2db0f2892aae58e4ffb1d4c/lib/shakapacker/instance.rb#L48).

Since you're trying to use `Procfile.dev-static`, that's not happening, which means that the `react-refresh` babel plugin shouldn't be being used.

Since it apparently was, my guess is that you have `webpack-dev-server` running either in the background or on another tab.

You can check for this by adding something like:
```
  if(process.env.WEBPACK_SERVE) {
    throw "process.env.WEBPACK_SERVE shouldn't be defined!!!"
  }
  ```
  to your `babel.config.js`.
